### PR TITLE
Adding overrides & flexibility to AdHocScheduler

### PR DESCRIPTION
**Pull request by jblayneyXpanxion**
Overview:
- Updated config to use explicit component scans, and a config package for bean definitions to allow for more creative integration test setups with mock beans.
- Added the ability to schedule jobs with parameters specified via JobConfig. 
- Added the ability to schedule jobs via name that are defined in a job bean that is constructed on-demand. This is helpful when job steps are defined as beans with parameters resolved via spel notation based off JobParameters/JobContext that is only available in the StepScope at job construction.
- Updated the default group to be Quartz's default group (specified as a null param to JobKey), rather than a hard-coded "group" string. 
- Wrote Integration tests to verify parameters are passed to the jobLauncher via this library's QuartzJobLauncher
- Bumped spring up to 2.2.6

### CODE REVIEW

**Comment by jblayneyXpanxion** [comment]
Had to refactor some config classes to add more test flexibility. The aspectJAutoProxy was causing some odd issues. 

I'll fix the remaining test coverage tomorrow, and I'm going to remove the TimeZone overloads. After learning more about how cron expressions work, time zone can be specified in the cron expression string, and is internally set via parsing the cron expression, so these parameters can conflict with certain cron expressions. Will be simpler & more cohesive to have fewer overload methods without added timezone params.

**Comment by jblayneyXpanxion** [comment]
@chrisgleissner this guy is ready for you!

**Comment by jblayneyXpanxion** [comment]
@chrisgleissner thanks for the quick turnaround on the last one. I'm hoping to hear from you soon because I have a timeline, but if I don't I'm sure I can make something work! No issues. I can turn around any changes pretty quick for you within the next two business days.

File: util/src/test/java/com/github/chrisgleissner/springbatchrest/util/core/AdHocStarterTest.java [hunk 1]
```diff
@@ -53,8 +53,8 @@ public void startWorks() throws InterruptedException {
                 propMap.put("foo", propertyValue++);
                 starter.start(jobToRun, true, propMap);
             }
-            assertThat(latch.await(2, SECONDS)).isTrue();
-            assertThat(propertyValues).hasSize(propertyValue);
+            assertThat(latch.await(3, SECONDS)).isTrue();
+            assertThat(propertyValues).hasSize(propertyValue); // TODO: Intermittent failures here - presumed timing issue
```
Comment (chrisgleissner) [inline util/src/test/java/com/github/chrisgleissner/springbatchrest/util/core/AdHocStarterTest.java:57]: I spotted this as well and will investigate it.

**Comment by chrisgleissner** [comment]
Thanks a lot for this epic merge request. Much appreciated and great work.

**Comment by chrisgleissner** [comment]
I released your work as 1.5.0 as there were a bunch of API extensions and it's definitely more than a bug fix :-) I also fixed the sporadic test errors you commented on in the merge request. Thanks again for your work, much appreciated.

**Pull request merged**